File: Medium/Python/2405. Optimal Partition of String.py
class Solution(object):
    def partitionString(self, s):
        start = 0
        end = 0
        count = 0

        while end < len(s):
            if s[end] + "" in s[start:end]:
                count += 1
                start = end
            end += 1
        
        if s[end - 1] + "" in s[start:end]:
            count += 1
        
        return count

File: Medium/Java/2405. Optimal Partition of String.java
class Solution {
    public int partitionString(String s) {
        int start = 0;
        int end = 0;
        int count = 0;

        while(end < s.length()){
            if(s.substring(start, end).contains(s.charAt(end) + "")){
                count++;
                start = end;
            }
            end++;
        }

        if(s.substring(start, end).contains(s.charAt(end - 1) + "")){
            count++;
        }

        return count;
    }
}
